cpp/data_structures/dynamic_tree/dynamic_tree.hpp

```diff
@@ -0,0 +1,414 @@
+                }
+
+                // in case not found
+                if(&target_index != NULL) {
+                    target_index = -1;
+                }
+                return false;
+            }
+
+            // sort children
+            // o(n log n) --> o(n²)
+            void sort(bool reverse = false){
+
+                if(reverse){
+                    std::sort(this->children.begin() , this->children.end() , 
+                        // comparison function
+                        [](DT_Node<t> a , DT_Node<t> b){
+                            return (a > b);
+                        }
+                    );
+                }
+                else{
+                    std::sort(this->children.begin() , this->children.end() , 
+                        // comparison function
+                        [](DT_Node<t> a , DT_Node<t> b){
+                            return (a < b);
+                        }
+                    );
+                }
+
+            }
+
+            // o(1) --> o(log n)
+            // get a copy of specific child , this copy not connected to the original tree
+            // in case target not found return will be a empty node
+            DT_Node<t> getCopy(std::string target_node_name){
+                
+                int index = -1;
+
+                // search for target
+                this->search(target_node_name,index);
+
+                // temp copy
+                DT_Node<t> copy_of_child;
+
+                // in case target found
+                if(index != -1){
+                    // set name & value using index
+                    copy_of_child.name  = this->children[index].name;
+                    copy_of_child.value = this->children[index].value;
+                }
+
+                return copy_of_child;
+            }
+
+            // o(log n) --> o(nodes * log n)
+            // remove child with specific name with his all children
+            bool removeChild(std::string target_node_name){
+
+                // search for target index
+                int index = -1;
+                search(target_node_name , index);
+
+                // call recursive remove of this child node
+                if(index != -1){
+                    children[index].removeChildren();
+                    children.erase(children.begin() + index);
+                    return true;
+                } 
+
+                return false;
+            }
+
+            // like removeChild function but this for all children 
+            void removeChildren(){
+                
+                // nested recursive removeChildren => remove all children of children
+                for(DT_Node<t> &child : this->children){
+                    child.removeChildren();
+                }
+
+                // last step remove children by setup a new empty vector 
+                this->children = std::vector<DT_Node<t>>();
+            }
+
+			// testing function
+			void print(){
```

Comment:
no print inside the code function, sorry this need to be a clean library

Reply:
yeah sure